src/mlpack/tests/main_tests/preprocess_scale_test.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file preprocess_scale_test.cpp
+ * @author Jeffin Sam
+ *
+ * Test mlpackMain() of preprocess_scale_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+#include <mlpack/core.hpp>
+static const std::string testName = "PreprocessScale";
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/methods/preprocess/preprocess_scale_main.cpp>
+
+#include "test_helper.hpp"
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+#include <cmath>
+
+using namespace mlpack;
+
+struct PreprocessScaleTestFixture
+{
+ public:
+  PreprocessScaleTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~PreprocessScaleTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(PreprocessScaleMainTest,
+                         PreprocessScaleTestFixture);
+
+arma::mat dataset = "-1 -0.5 0 1;"
+                    "2 6 10 18;";
+/**
+ * Check that two different scalers give two different output.
+ */
+BOOST_AUTO_TEST_CASE(TwoScalerTest)
+{
+  std::string method = "max_abs_scaler";
+  // Input custom data points.
+  SetInputParam("input", dataset);
+  SetInputParam("scaler_method", std::move(method));
```

Comment:
Actually if you wanted to simplify this, you could just put `method` in `SetInputParam()` directly: `SetInputParam("scaler_method", "max_abs_scaler")`.  Up to you.

Reply:
`SetInputParam("scaler_method", "max_abs_scaler")` throws error and hence i haven't changed it.

Reply:
Oh, you're right, `SetInputParam()` takes an rvalue reference for the second argument, so I guess `std::move()` is needed.  It doesn't really make a difference either way. :)